src/app/routes/getInitialData/index.js

```diff
@@ -6,19 +6,48 @@ import onClient from '#lib/utilities/onClient';
 import getBaseUrl from './utils/getBaseUrl';
 import getPreprocessorRules from './utils/getPreprocessorRules';
 
-const logger = nodeLogger(__filename);
+export const logger = nodeLogger(__filename);
 const STATUS_CODE_OK = 200;
 const STATUS_CODE_BAD_GATEWAY = 502;
 const STATUS_CODE_NOT_FOUND = 404;
 const upstreamStatusCodesToPropagate = [STATUS_CODE_OK, STATUS_CODE_NOT_FOUND];
+const validRendererEnvironments = ['int', 'test', 'live'];
 
 const ampRegex = /(.amp)$/;
 
 const baseUrl = onClient()
   ? getBaseUrl(window.location.origin)
   : process.env.SIMORGH_BASE_URL;
 
-const getUrl = pathname => `${baseUrl}${pathname.replace(ampRegex, '')}.json`;
+// eslint-disable-next-line consistent-return
+export const getUrl = pathname => {
+  if (pathname) {
+    const regex = /renderer_env=(.*)&?/i;
+    const [match, environment] = pathname.match(regex) || [];
```

Comment:
We have Ramda in simorgh - ramda's `match` function returns an empty array when there are no matches - would be an option here to remove the `||` and associated cyclomatic complexity hit in code climate

https://ramdajs.com/docs/#match

Reply:
Cheers, will have a look

Reply:
Using querystringify, regex is no longer needed :)